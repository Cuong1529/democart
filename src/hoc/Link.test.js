import renderer from "react-test-renderer";
import Link from "./Link";

it("change", () => {
  const component = renderer.create(<Link page="face.com">Face</Link>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
