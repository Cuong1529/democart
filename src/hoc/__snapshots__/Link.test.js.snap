// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change 1`] = `
<a
  className="normal"
  href="face.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Face
</a>
`;

exports[`change 2`] = `
<a
  className="hovered"
  href="face.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Face
</a>
`;

exports[`change 3`] = `
<a
  className="normal"
  href="face.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Face
</a>
`;
